### ISSUE TITLE
microk8s.enable fluentd not sucessful

### ISSUE THREAD

**Issue by dannyyy** [ubuntu/microk8s#891]
I'm running the Ubuntu 19.10 Server release. Microk8s has been installed as part of the OS installation. During the setup the installer asked for snap packes to install.

I tried to enable some of the addins. Most of them haven't raised any errors (haven't tested the functionallity all of them). But trying to install the fluentd addin raises the following error:

```
Enabling Fluentd-Elasticsearch
Labeling nodes
error: 'beta.kubernetes.io/fluentd-ds-ready' already has a value (true), and --overwrite is false
Enabling DNS
Applying manifest
serviceaccount/coredns unchanged
configmap/coredns unchanged
deployment.apps/coredns unchanged
service/kube-dns unchanged
clusterrole.rbac.authorization.k8s.io/coredns unchanged
clusterrolebinding.rbac.authorization.k8s.io/coredns unchanged
Restarting kubelet
DNS is enabled
Unable to connect to the server: dial tcp 127.0.0.1:16443: i/o timeout
Failed to enable fluentd
```

And sometome
```
Enabling Fluentd-Elasticsearch
Labeling nodes
node/df-k8s-mikro labeled
Enabling DNS
Applying manifest
serviceaccount/coredns created
configmap/coredns created
deployment.apps/coredns created
service/kube-dns created
clusterrole.rbac.authorization.k8s.io/coredns created
clusterrolebinding.rbac.authorization.k8s.io/coredns created
Restarting kubelet
DNS is enabled
Unable to connect to the server: net/http: TLS handshake timeout
Failed to enable fluentd
```

At the beginning I had sometimes this (`microk8s.kubectl describe node`)
```
Warning  MissingClusterDNS        6m6s (x5 over 6m6s)  kubelet, df-k8s-mikro     kubelet does not have ClusterDNS IP configured and cannot create Pod using "ClusterFirst" policy. Falling back to "Default" policy.
Warning  InvalidDiskCapacity      2m10s                kubelet, df-k8s-mikro     invalid capacity 0 on image filesystem
```

In the meantime I only have this
```
Warning  InvalidDiskCapacity      2m10s                kubelet, df-k8s-mikro     invalid capacity 0 on image filesystem
```

[inspection-report-20200103_171200.tar.gz](https://github.com/ubuntu/microk8s/files/4020105/inspection-report-20200103_171200.tar.gz)

**Comment by balchua** [comment]
When enabling the DNS, it will restart the kubelet and then waits for about 5 seconds then reconfigure the api-server to allow-privileged pods.  sleep for 5 seconds then install the fluentd addons.
My guess is that the api-server has not fully started before installing the fluentd addons.

Maybe we should wait a bit more than just 5 seconds, or do a retry to check if api-server is up and running before proceeding.

**Comment by dannyyy** [comment]
@balchua 
I installed the DNS server as one of the very first after setting up Microk8s. In between there must be minutes. But when the `fluentd` addin will be installed, it automatically triggers again the DNS installation. Maybe it would be smarter to only do this if `microk8s.status` say DNS is disabled.

In the meantime I was able to install it (properly). I had to restart microk8s with `.stop` / `.start`. After few minutes the installation is still failing even the API seems up and running. After what seemed like ages (at lest 15-20min) I tried it again and voila the installation procedure was successful.

I was a bit astonished about the behaviour and wiped the disk and did a proper re-installation of Ubuntu including Microk8s. And guess what, exactly the same behaviour. Approx. 20min after starting or restarting Microk8s (OS restart or service restart) I was able the first time again, to install it. Before I got the exact same errors.

Either it has to do, that the script wants to reinstall DNS including same API changes which are not ready (but why after 20min) or the mini-cluster is in a healing/recovery state für such a long time after a restart. But I'm a very novice in operating Kubernetes and don't know exactly what's going on under the hood.
